fmriprep/interfaces/cifti.py

```diff
@@ -0,0 +1,282 @@
+        """Assign arbitrary label to combination of CIFTI spaces"""
+        space = None
+        variants = {
+            # to be expanded once addtional spaces are supported
+            'space1': ['fsaverage5', 'MNI152NLin2009cAsym'],
+            'space2': ['fsaverage6', 'MNI152NLin2009cAsym'],
+        }
+        for sp, targets in variants.items():
+            if all(target in targets for target in
+                    [self.inputs.surface_target, self.inputs.volume_target]):
+                space = sp
+
+        if space is None:
+            raise NotImplementedError
+        return space
+
+    def _fetch_data(self):
+        """Converts inputspec to files"""
+        if (self.inputs.surface_target == "fsnative" or
+                self.inputs.volume_target != "MNI152NLin2009cAsym"):
+            # subject space is not support yet
+            raise NotImplementedError
+
+        annotation_files = sorted(glob(os.path.join(self.inputs.subjects_dir,
+                                                    self.inputs.surface_target,
+                                                    'label',
+                                                    '*h.aparc.annot')))
+        if not annotation_files:
+            raise IOError("Freesurfer annotations for %s not found in %s" % (
+                          self.inputs.surface_target, self.inputs.subjects_dir))
+
+        label_space = 'oasis_dkt31_mni152'
+        label_template = os.path.dirname(getters.get_dataset(label_space))
+        label_file = os.path.join(label_template,
+                                  'oasis_trt20_DKT31-MNI152NLin',
+                                  ('OASIS-TRT-20_jointfusion_DKT31_CMA_labels'
+                                   '_in_MNI152NLin2009cAsym_2mm_v2.nii.gz'))
+
+        download_link = getters.OSF_PROJECT_URL + getters.OSF_RESOURCES[label_space][0]
+        return annotation_files, label_file, download_link
+
+    @staticmethod
+    def _create_cifti_image(bold_file, label_file, annotation_files, gii_files,
+                            volume_target, surface_target, tr, download_link=None):
+        """
+        Generate CIFTI image in target space
+
+        Parameters
+            bold_file : 4D BOLD timeseries
+            label_file : label atlas
+            annotation_files : FreeSurfer annotations
+            gii_files : 4D BOLD surface timeseries in GIFTI format
+            volume_target : label atlas space
+            surface_target : gii_files space
+            tr : repetition timeseries
+            download_link : URL to download label_file
+
+        Returns
+            out_file : BOLD data as CIFTI dtseries
+        """
+
+        bold_img = nb.load(bold_file)
+        label_img = nb.load(label_file)
+        if not bold_img.shape[:3] == label_img.shape:
+            # we need these images to have the same resolution
+            bold_img = resample_to_img(bold_img, label_img, interpolation="nearest")
```

Comment:
Is the image resampled this way then used when sampling on the surface? (in other words, are we not using the promised Lanczos interpolation on an important resampling?)

Reply:
@oesteban I don't think this shouldn't affect the surfaces, since those are acquired from the GIFTI files. This resampling is only done to ensure the subcortical structures can be indexed by the same coordinates found in the atlas.

Reply:
Sure, so basically you need to pack ((i, j, k, t), bold_value) within the format right? Thus, that bold_value will be the result of this interpolation. Correct?

Reply:
^ yes, but only for subcortical - I believe surface voxels (i,j,k) are stored via GIFTI.

Reply:
Okay, I'll have to have a deeper look into this. If I can't check on it anytime soon, I'll open an issue and merge this as is :+1:

Reply:
@oesteban, did you get a chance to think about this? This is *hopefully* the last thing before merging

Reply:
I don't have time dive into this currently, so I just opened an issue for the future :+1: . I think this is good to go!

Reply:
Okay, we had a discussion over this line, and we think it should be fine if you just drop the `interpolation='nearest'`. The issue is that in many cases this is going to be upsampling BOLD data to 2mm^3 voxels, which should really be interpolated.

Reply:
forgive my ignorance, but how is continuous interpolation applied? I understand linear and nearest

Reply:
Cubic spline interpolation. It sets `order=3` in [`ndimage.affine_transform()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.affine_transform.html).